### PR TITLE
Add GHCB MSR protocol unit tests

### PR DESCRIPTION
**Pull request by conradgrobler**
This PR adds unit tests for the conversion of the MSR protocol request objects to u64 values and u64 values to response objects.

See section 2.3.1 of [SEV-ES Guest-Hypervisor Communication Block Standardization](https://developer.amd.com/wp-content/resources/56421.pdf) for the specification.

### CODE REVIEW

File: experimental/sev_guest/src/msr.rs [hunk 1]
```diff
@@ -462,3 +462,174 @@ pub fn read_msr() -> u64 {
...
+            (valid_page as u64) | ((assignment as u64) << 52) | 0x14,
+            msr_value
+        );
+    }
+
+    #[test]
+    fn test_snp_page_state_change_response() {
+        let error_code = 1u64;
+        let invalid = 0x15u64 | (error_code << 31);
+        let invalid2 = 0x16u64 | (error_code << 32);
```
Comment (rbehjati) [inline experimental/sev_guest/src/msr.rs:595]: Is it possible to give it a better name?

**Review by rbehjati — APPROVED** [review]
Nice. Thanks. I feel like generally some documentation would be nice. In particular, it is not clear to me where the different values come from, particularly for `page`. I assume the `page` values are not any arbitrary value. It also feels like the tests are mostly validating the conversion from numbers to various Request and Response types. If that is the intention, perhaps this could be described as a comment in `mod tests`.  

(Or maybe everything is clear to someone who is more familiar with SEV :))

File: experimental/sev_guest/src/msr.rs (hunk 1 above)
Comment (conradgrobler) [reply experimental/sev_guest/src/msr.rs]: Done.

**Review by conradgrobler — COMMENTED** [review]
Thanks for the review.

Yes, the tests only check the conversions. I added a comment to the `tests` module to clarify.

The values (e.g. page addresses) in the tests are arbitrary, they just need to fall in certain ranges to be considered valid for the protocol. Any testing of the actual protocol would require running inside a VM under AMD SEV-SNP.

**Pull request merged**